drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -360,7 +360,8 @@ static int _set(netdev_t *netdev, netopt_t opt, void *val, size_t len)
     }
 
     /* temporarily wake up if sleeping */
-    if (old_state == AT86RF2XX_STATE_SLEEP) {
+    if ((old_state == AT86RF2XX_STATE_SLEEP) &&
+        (opt != NETOPT_STATE)) {
```

Comment:
Also please add a comment that the the `opt != NETOPT_STATE` check prevents redundant wake-up (also maybe hinting that for that `NETOPT` the state changes anyway... had to look that up again myself ;-))